hmp/models/fixedn.py

```diff
@@ -727,25 +659,23 @@ def estim_probs(
         eventprobs : ndarray
             Probabilities with shape max_samples*n_trials*n_events
         """
-        if n_events is None:
-            n_events = magnitudes.shape[0]
+        n_events = magnitudes.shape[0]
         n_stages = n_events + 1
-
+        if location:
+            locations = np.zeros(n_stages, dtype=int)
+            locations[1:-1] = self.location
+        else:
+            locations = np.zeros(n_stages, dtype=int)
         if subset_epochs is not None:
             if len(subset_epochs) == trial_data.n_trials:  # boolean indices
                 subset_epochs = np.where(subset_epochs)[0]
-            n_trials = len(subset_epochs)
-            durations = trial_data.durations[subset_epochs]
-            starts = trial_data.starts[subset_epochs]
-            ends = trial_data.ends[subset_epochs]
-        else:
-            n_trials = trial_data.n_trials
-            durations = trial_data.durations
-            starts = trial_data.starts
-            ends = trial_data.ends
-
+        n_trials = len(subset_epochs)
+        durations = trial_data.durations[subset_epochs]
+        starts = trial_data.starts[subset_epochs]
+        ends = trial_data.ends[subset_epochs]
+        max_duration = np.max(durations)
         gains = np.zeros((trial_data.n_samples, n_events), dtype=np.float64)
-        for i in range(trial_data.n_dims):
+        for i in range(trial_data.cross_corr.shape[1]):
```

Comment:
Why has this changed?

Reply:
In most cases n_dim is part of model. Here is the only case where we actually need it from the data so I thought it can be read from the shape

Reply:
Okay, makes sense. Though perhaps if can be a property of TrialData?

Reply:
see commit [b8765c5](https://github.com/qubixes/hmp/pull/8/commits/b8765c5cf95f2225b819506554ec881f6a0b24b5)

Reply:
Ah, I see you made it back into an attribute. I meant a property, so that you don't have to give it to the TrialData object, but will be automatically calculated based on the cross_corr values (`@property`). You can look at the `cached_property` below for an example. (Here, caching is probably not very useful.)

Reply:
should be good now